Hope I test it correctly, but had this error during validation before publishing.
```
DEBUG: Looking for matching profile for: host_names: "nuke" | product_base_types: "render" | task_names: "GENERIC" | task_types: "Generic"
DEBUG: "nuke" not found in "host_names": ['substancedesigner']
DEBUG: "nuke" not found in "host_names": ['maya']
DEBUG: "nuke" not found in "host_names": ['traypublisher']
DEBUG: "nuke" not found in "host_names": ['max']
DEBUG: "nuke" not found in "host_names": ['unreal']
DEBUG: Profile selected: {'host_names': [], 'product_base_types': ['review', 'render', 'prerender'], 'task_names': [], 'task_types': [], 'template_name': 'render'}
Traceback (most recent call last):
  File "C:\Users\mila\AppData\Local\Ynput\AYON\dependency_packages\ayon_2602231726_windows.zip\dependencies\pyblish\plugin.py", line 528, in __explicit_process
    runner(*args)
  File "C:\Users\mila\AppData\Local\Ynput\AYON\addons\core_1.9.0+dev\ayon_core\plugins\publish\collect_resources_path.py", line 111, in process
    original_directory = instance.data["stagingDir"]
                         ~~~~~~~~~~~~~^^^^^^^^^^^^^^
KeyError: 'stagingDir'
```

<img width="1406" height="569" alt="image" src="https://github.com/user-attachments/assets/aadb30e7-ce64-4ecb-8552-c7207aa70815" />

--- a/client/ayon_core/plugins/publish/collect_resources_path.py
+++ b/client/ayon_core/plugins/publish/collect_resources_path.py
@@ -13,7 +13,7 @@ import copy
 
 import pyblish.api
 
-from ayon_core.pipeline.publish import get_publish_template_name
+from ayon_core.pipeline.publish import get_publish_template_name, PublishError
 
 
 class CollectResourcesPath(pyblish.api.InstancePlugin):
@@ -109,13 +109,14 @@ class CollectResourcesPath(pyblish.api.InstancePlugin):
             original_directory = instance.data.get("originalDirname")
             if not original_directory:
                 original_directory = instance.data.get("stagingDir")
-                if not original_directory:
-                    self.log.warning(
-                        "Publish template requires 'originalDirname'"
-                        " but 'originalDirname' is not set on instance"
-                        " and 'stagingDir' is not yet filled."
-                    )
-                    return
+
+            if not original_directory:
+                raise PublishError(
+                    "Publish template requires 'originalDirname'"
+                    " but 'originalDirname' is not set on instance"
+                    " and 'stagingDir' is not yet filled."
+                )
+
             template_data["originalDirname"] = original_directory
 
         publish_folder = os.path.normpath(

--- a/client/ayon_core/plugins/publish/collect_resources_path.py
+++ b/client/ayon_core/plugins/publish/collect_resources_path.py
@@ -108,7 +108,14 @@ class CollectResourcesPath(pyblish.api.InstancePlugin):
         if "{originalDirname}" in publish_template:
             original_directory = instance.data.get("originalDirname")
             if not original_directory:
-                original_directory = instance.data["stagingDir"]
+                original_directory = instance.data.get("stagingDir")
+                if not original_directory:
+                    self.log.warning(
+                        "Publish template requires 'originalDirname'"
+                        " but 'originalDirname' is not set on instance"
+                        " and 'stagingDir' is not yet filled."
+                    )
+                    return
             template_data["originalDirname"] = original_directory
 
         publish_folder = os.path.normpath(

